File: solutions/ruby/high-scores/1/high_scores.rb
class HighScores
  attr_accessor :scores

  def initialize(scores)
    @scores = scores
  end

  def latest
    @scores[-1]
  end

  def personal_best
    @scores.sort[-1]
  end

  def personal_top_three
    @scores.size < 3 ? @scores.sort.reverse : @scores.sort[-3..-1].reverse
  end

  def latest_is_personal_best?
    latest == personal_best
  end
end

File: solutions/cpp/high-scores/1/high_scores.cpp
#include "high_scores.h"

#include <algorithm>
#include <functional>

namespace arcade {
std::vector<int> HighScores::list_scores() const {
    return scores;
}

int HighScores::latest_score() const {
    return scores.back();
}

int HighScores::personal_best() const {
    return *(std::minmax_element(scores.begin(), scores.end()).second);
}

std::vector<int> HighScores::top_three() const {
    auto result = std::vector<int>(scores);
    std::sort(result.begin(), result.end(), [](int a, int b){return a>b;});
    if (result.size() > 3) result.resize(3);
    return result;
}

}  // namespace arcade
